src/tracker/engaged_time.js

```diff
@@ -85,9 +85,19 @@ limitations under the License.
         root.focused = false;
     });
 
+    var supportsPassive = false;
+    try {
+        addEventListener("test", null, Object.defineProperty({}, 'passive', {get: function () {
```

Comment:
Yeah it's a little ugly / confusing, sorry about that.  Basically the canonical way (only way at the moment) to feature detect for dictionary members is to pass a custom object to an API that takes the dictionary and see if the getter is invoked for the property of interest.  [WebIDL Ecmascript bindings](http://heycam.github.io/webidl/#es-dictionary) require that the getter for each supported property is invoked when the function is called.

So in this case we do a dummy call to `addEventListener` (passing null as the handler is defined to be a no-op).  We pass into that a custom object whose `passive` getter function just records that it has been called.  There are four possible behaviors:

1. Browser doesn't support `addEventListener` or `defineProperty` at all - throws, `supportsPassive` remains false.
2. `addEventListener` expects only a boolean in the 3rd position.  The object gets coerced to the value `true` without invoking the getter.  `supportsPassive` remains false.
3. `addEventListener` supports taking a dictionary in the 3rd position, but not the `passive` member.  In that case other getters may be invoked, but not the one we've defined for `passive`.  `supportsPassive` remains false.
4. `addEventListener` supports the passive option.  In that case the getter function we've defined gets run as part of the JavaScript API binding and `supportsPassive` will be true.

For context see the [original design discussion](https://github.com/WICG/EventListenerOptions/issues/16#issuecomment-120064955) and [this WebIDL issue](https://github.com/heycam/webidl/issues/107) where we're considering exposing something less ugly.